### DISCUSSION TITLE
Why fork

### DISCUSSION THREAD

**Discussion by woj-tek** [XilinJia/Podcini#34] [General]
Why fork and not contribute back to AntennaPod?

**Comment by XilinJia** [comment]
Well, interesting question.  I guess for multiple reasons.

At first, when it could be easy to merge in, I tried to see if there was any interest in my upgrading to media3, not much.

Then as I looked through the Java code, to me there appear quite some hard-to-control null possibilities to easily do further coding.  So I started converting the whole thing to Kotlin.  And indeed, the conversion process manifested quite some issues that I took effort to resolve.  So at this point I expected much less interest to merge (actually there is basically nothing to merge, but a re-base). 

Then I wanted to change the subscriptions view (I don't like tags being folders), and I don't see much need for the home screen, etc.  And I saw the project structure is stretched with multiple modules.  So I did quite some drastic changes, starting with version 4.

Then if you check the changelog file, I've made lots of further changes up so far.

So, to answer your question, it's for a podcast player to my and others' liking, the flexibility to make bigger changes, the fast pace from ideas to released apk, and the impossibility to merge.

**Reply by woj-tek** [reply]
Thanks for the very detailed and in depth explanation. Indeed, it makes sense :-)
